packages/fast-tooling-react/src/css-property-editor/property-editor-row.tsx

```diff
@@ -0,0 +1,230 @@
+        return `${get(this.props.managedClasses, "cssPropertyEditorRow_input", "")} ${get(
+            this.props.managedClasses,
+            "cssPropertyEditorRow_inputKey",
+            ""
+        )}`;
+    }
+
+    /**
+     * get classes for value input
+     */
+    private generateValueClassNames(): string {
+        return `${get(this.props.managedClasses, "cssPropertyEditorRow_input", "")} ${get(
+            this.props.managedClasses,
+            "cssPropertyEditorRow_inputValue",
+            ""
+        )}`;
+    }
+
+    /**
+     * value has changed in key input
+     */
+    private handleKeyInputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        const newKey: string = e.target.value;
+        if (
+            typeof this.props.onKeyChange === "function" &&
+            this.props.cssKey !== newKey
+        ) {
+            this.props.onKeyChange(
+                camelCase(newKey),
+                this.props.cssKey,
+                this.props.rowIndex
+            );
+        }
+    };
+
+    /**
+     * value has changed in value input
+     */
+    private handleValueInputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        const newValue: string = e.target.value;
+        if (
+            typeof this.props.onValueChange === "function" &&
+            this.props.value !== newValue
+        ) {
+            this.props.onValueChange(newValue, this.props.cssKey, this.props.rowIndex);
+        }
+    };
+
+    /**
+     *  Handle focus events
+     */
+    private handleFocus = (): void => {
+        if (typeof this.props.onRowFocus === "function") {
+            this.props.onRowFocus(this.props.cssKey, this.props.rowIndex);
+        }
+    };
+
+    /**
+     *  key input has lost focus
+     */
+    private handleKeyInputBlur = (e: React.FocusEvent<HTMLInputElement>): void => {
+        if (typeof this.props.onCommitKeyEdit === "function") {
+            this.props.onCommitKeyEdit(this.props.cssKey, this.props.rowIndex);
+        }
+
+        this.checkForRowBlur(e);
+    };
+
+    /**
+     *  value input has lost focus
+     */
+    private handleValueInputBlur = (e: React.FocusEvent<HTMLInputElement>): void => {
+        this.checkForRowBlur(e);
+    };
+
+    /**
+     *  check if the row is losing focus in the provided focus event
+     */
+    private checkForRowBlur = (e: React.FocusEvent<HTMLInputElement>): void => {
+        if (
+            typeof this.props.onRowBlur === "function" &&
+            !isNil(this.rowRef.current) &&
+            !this.rowRef.current.contains(e.relatedTarget as Element)
+        ) {
+            this.props.onRowBlur(this.props.cssKey, this.props.rowIndex);
+        }
+    };
+
+    /**
+     * Handle click
+     */
+    private handleClick = (e: React.MouseEvent<HTMLDivElement>): void => {
+        if (
+            typeof this.props.onClickToInsert === "function" &&
+            (e.target as HTMLElement).nodeName !== "INPUT"
```

Comment:
what if the click target is any other input on the page?

Reply:
This is attached to the row div so it should only respond to clicks within the row.